src/project-manager.ts

```diff
@@ -735,6 +754,11 @@ export class ProjectConfiguration {
 	private expectedFilePaths = new Set<string>();
 
 	/**
+	 * List of resolved extra root directories to allow global type declaration files to be loaded from.
```

Comment:
Is this restricted to global typings?

Reply:
From https://www.typescriptlang.org/docs/handbook/tsconfig-json.html:
> By default all visible “@types” packages are included in your compilation.
> If typesRoots is specified, only packages under typeRoots will be included.
> Keep in mind that automatic inclusion is only important if you’re using files with global declarations (as opposed to files declared as modules)

Reply:
They clarify in that page as well: 
> If you use an import "foo" statement, for instance, TypeScript may still look through node_modules & node_modules/@types folders to find the foo package.